### PR TITLE
Examples for using custom integrators

### PR DESCRIPTION
**Pull request by kannandeepti**
- Added two example scripts in examples/customIntegrators that illustrate how to use the custom integrators in contrib/integrators.py
- Added more documentation contrib/integrators.py
- modified "integrator" parameter in Simulation class constructor: now takes in a tuple (integrator, integrator_type) containing the openMM class reference and a string specifying the integrator type. Before, the integrator_type class attribute was set to "UserDefined". Now the user can specify a different identifier. For example, both of my integrators are brownian integrators, so I would pass the Simulation class `integrator=(integrator, "brownian")`. This will ensure that an error is not raised if the polymer's kinetic energy exceeds eK_critical (this check is not relevant for Brownian dynamics, since kinetic energy is somewhat meaningless when monomers do not have inertia). Similarly if a user writes their own "variableLangevin" integrator, they can specify the integratory_type as "variableLangevin" so that the Simulation class includes the time stepping information in the reporter.

### CODE REVIEW

File: polychrom/simulation.py [hunk 1]
```diff
@@ -743,7 +745,7 @@ def do_block(
                 sys.stdout.flush()
             self._apply_forces()
             self.forces_applied = True
-
+
```
Comment (mimakaev) [inline polychrom/simulation.py:748]: can you apply "black" to the codebase? It will get rid of stray spaces etc.

File: scripts/hot_cold.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+from simtk import unit
+from pathlib import Path
+
+total_runs = 200
+runs_per_gpu = total_runs // 4
+
+def run_sim(gpuid, run_number, activity_ratio, timestep=170, ntimesteps=200000, blocksize=100):
+    """ Run a single simulation on GPU i."""
+    ids = np.load('/net/levsha/share/deepti/data/ABidentities_blobel2021_chr2_35Mb_60Mb.npy')
+    N=len(ids)
```
Comment (mimakaev) [inline scripts/hot_cold.py:24]: I would consider adding these files to the repository in some examples folder. They won't work for people who are not in the mirny lab.

**Comment by mimakaev** [comment]
please remove permission changes to files - permissions should be 644, not 755, for Python files

File: scripts/hot_cold.py (hunk 2 above)
Comment (kannandeepti) [reply scripts/hot_cold.py]: whoops this file wasn't supposed to be part of the PR. removed it

File: polychrom/simulation.py (hunk 1 above)
Comment (kannandeepti) [reply polychrom/simulation.py]: did so with default settings, including on files not part of this PR

**Comment by mimakaev** [comment]
wow that's a lot of black!

File: polychrom/rouse.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+""" 
+Simulation parameters
+---------------------
+
+This module provides some intuition for the units that are used in polychrome.
+For example, the mass of a particle is automatically set to 100 amu and 
+the collision rate is set depending on the integrator. For Brownian integrators,
+setting the collision rate to 2.0 works well, whereas for Langevin integrators,
+it is best to set the collision rate to 0.001-0.01. For loop extrusion simulations,
```
Comment (mimakaev) [inline polychrom/rouse.py:9]: Did you intend to add this file?

**Comment by mimakaev** [comment]
I'm happy to merge this with all the black.

**Review by mimakaev — APPROVED** [review]

**Pull request merged**